### PR TITLE
Enable Buildifier in CI

### PR DESCRIPTION
**Pull request by UebelAndre**
The purpose of this PR is to enable Buildifier in CI. This is done by mostly ignoring existing defects but still allows for future changes to benefit from this check.

New changes should not "disable" any defects.

### CODE REVIEW

**Review by jsharpe — APPROVED** [review]

**Comment by attilaolah** [comment]
It might be neater to do this in two steps:

1. First just run `buildifier` to fix formatting / whitespace-related issues.
2. Then run with `-lint=fix` to do the more substantial changes, like replacing `navice` rules with the Starlark ones from `@rules_cc`.

Note also that CI is failing in `'//cmake_android:app'`.

File: examples/cmake_android/BUILD [hunk 1]
```diff
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+load("@rules_android//android:rules.bzl", "android_binary", "android_library")
```
Comment (attilaolah) [inline examples/cmake_android/BUILD:2]: This seems to be [failing on BuildKite](https://buildkite.com/bazel/rules-foreign-cc/builds/1282#0428871d-995f-45d2-8b16-662c8c404077):

```
(14:10:54) ERROR: error loading package 'cmake_android': Unable to find package for @rules_android//android:rules.bzl: The repository '@rules_android' could not be resolved.
```

**Comment by jsharpe** [comment]
I'm actually not sure that depending on `rules_cc` is the right thing to do yet as the transition path to starlark cc rules isn't yet well defined. We've actually done what you suggested already and merged a PR to fix up just the formatting without lint fixes.

**Comment by UebelAndre** [comment]
Yeah, this has been mostly solved for in https://github.com/bazelbuild/rules_foreign_cc/pull/459 but I want to see what else we can do here so users can freely run buildifier on this repo and not run into any issues.

**Comment by UebelAndre** [comment]
Ok, this is in much better shape. I'll split out some PRs from this but feel free to take a look here and let me know what you think. @attilaolah @jsharpe 

> I'm actually not sure that depending on `rules_cc` is the right thing to do yet as the transition path to starlark cc rules isn't yet well defined.

I think the intent that the `rules_cc` repo be used is pretty clear. Most (if not all) the symbols in that repo are wrappers around the native call so this shouldn't be a big change. Additionally, *all* uses of `rules_cc` are in examples so there's very little risk to introducing it. Do you concur?

File: examples/cmake_android/BUILD (hunk 1 above)
Comment (UebelAndre) [reply examples/cmake_android/BUILD:1]: fixed

**Comment by jsharpe** [comment]
> Ok, this is in much better shape. I'll split out some PRs from this but feel free to take a look here and let me know what you think. @attilaolah @jsharpe
> 
> > I'm actually not sure that depending on `rules_cc` is the right thing to do yet as the transition path to starlark cc rules isn't yet well defined.
> 
> I think the intent that the `rules_cc` repo be used is pretty clear. Most (if not all) the symbols in that repo are wrappers around the native call so this shouldn't be a big change. Additionally, _all_ uses of `rules_cc` are in examples so there's very little risk to introducing it. Do you concur?

That was the original intent for `rules_cc` but the original migration project failed. See https://docs.google.com/document/d/1GW7UVo1s9X0cti9OMgT3ga5ozKYUWLPk9k8c4-34rC4/edit# for the current thinking on how the migration should proceed. Also see issue https://github.com/bazelbuild/rules_cc/issues/92.

However as you say with all uses being in the examples repo then this doesn't introduce an extra dependency on the end user and so if it gets us to the goal of being able to run Buildifier in lint mode as a CI check then I'm for it. 👍

**Comment by UebelAndre** [comment]
Huzzah! This PR is now ready 😄

**Review by jsharpe — CHANGES_REQUESTED** [review]
Can we put the docs update into a separate PR?

**Comment by UebelAndre** [comment]
> Can we put the docs update into a separate PR?

Yeah: https://github.com/bazelbuild/rules_foreign_cc/pull/473

**Review by jsharpe — APPROVED** [review]

**Pull request merged**